### PR TITLE
Added few method to edit Adaptive Frequency Strategy in Hfss

### PR DESCRIPTION
**Pull request by maxcapodi78**
enable_adaptive_setup_multifrequency
enable_adaptive_setup_broadband
enable_adaptive_setup_single

### CODE REVIEW

File: pyaedt/modules/SolveSetup.py [hunk 1]
```diff
@@ -538,6 +538,112 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+            self._app.logger.error("Method applies only to Hfss Driven Solutions.")
+            return False
+        self.auto_update = False
+        self.props["SolveType"] = "BroadBand"
+        for el in list(self.props["MultipleAdaptiveFreqsSetup"].keys()):
+            del self.props["MultipleAdaptiveFreqsSetup"][el]
+        if isinstance(low_frequency, (int, float)):
+            low_frequency = "{}GHz".format(low_frequency)
+        if isinstance(high_frquency, (int, float)):
+            high_frquency = "{}GHz".format(high_frquency)
```
Comment (jeremich41) [inline pyaedt/modules/SolveSetup.py:603]: high_frequency line 603 and 605

File: pyaedt/modules/SolveSetup.py [hunk 2]
```diff
@@ -538,6 +538,112 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        i = 0
+        for f in frequencies:
+            if isinstance(max_delta_s, float):
+                if isinstance(f, (int, float)):
+                    f = "{}GHz".format(f)
+                self.props["MultipleAdaptiveFreqsSetup"][f] = [max_delta_s]
+            else:
+                if isinstance(f, (int, float)):
+                    f = "{}GHz".format(f)
+                self.props["MultipleAdaptiveFreqsSetup"][f] = [max_delta_s[i]]
```
Comment (jeremich41) [inline pyaedt/modules/SolveSetup.py:642]: if i < len(max_delta_s) - 1:
       self.props["MultipleAdaptiveFreqsSetup"][f] = [max_delta_s[i]]
else:
       self.props["MultipleAdaptiveFreqsSetup"][f] = [0.02]

Comment (maxcapodi78) [reply pyaedt/modules/SolveSetup.py]: implemented with Try/Except

**Review by jeremich41 — APPROVED** [review]

File: pyaedt/modules/SolveSetup.py [hunk 3]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
         self.auto_update = legacy_update
         return True
 
+    @pyaedt_function_handler()
+    def enable_adaptive_setup_single(self, freq=None, max_passes=None, max_delta_s=None):
+        """Enable Hfss Single Frequency Setup.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:543]:
```suggestion
        """Enable HFSS single frequency setup.
```

File: pyaedt/modules/SolveSetup.py [hunk 4]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+
+        Parameters
+        ----------
+        freq : float, str
+            Frequency at which set the adaptive convergence. It can be float (GHz) or str.
+        max_passes : int
+            Maximum number of adaptive passes.
+        max_delta_s : float
+            Delta S Convergence criteria.
```
[the remark this answers is not in the document]
Comment (PipKat) [reply pyaedt/modules/SolveSetup.py:552]: These are all cases where the behavior resulting when the default of ``None`` is used should be explained: "The default is ``None``, in which case the ...."

File: pyaedt/modules/SolveSetup.py [hunk 5]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+
+        Returns
+        -------
+        bool
+        """
+        if self.setuptype != 1:
+            self._app.logger.error("Method applies only to Hfss Driven Solutions.")
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:559]:
```suggestion
            self._app.logger.error("Method applies only to HFSS-driven solutions.")
```

File: pyaedt/modules/SolveSetup.py [hunk 6]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        if max_passes:
+            self.props["MaximumPasses"] = max_passes
+        if max_delta_s:
+            self.props["MaxDeltaS"] = max_delta_s
+        self.auto_update = True
+        return self.update()
+
+    @pyaedt_function_handler()
+    def enable_adaptive_setup_broadband(self, low_frequency, high_frquency, max_passes=6, max_delta_s=0.02):
+        """Enable Hfss Broadband Setup.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:576]:
```suggestion
        """Enable HFSS broadband setup.
```

File: pyaedt/modules/SolveSetup.py [hunk 7]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        Parameters
+        ----------
+        low_frequency : float, str
+            Lower Frequency at which set the adaptive convergence. It can be float (GHz) or str.
+        high_frquency : float, str
+            Lower Frequency at which set the adaptive convergence. It can be float (GHz) or str.
+        max_passes : int
+            Maximum number of adaptive passes.
+        max_delta_s : float
+            Delta S Convergence criteria.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:588]:
```suggestion
        low_frequency : float, str
            Lower frequency at which to set the adaptive convergence. You can enter a float value
            in GHz or a string.
        high_frquency : float, str
            Lower frequency at which to set the adaptive convergence. You can enter a float value
            in GHz or a string.
        max_passes : int, optional
            Maximum number of adaptive passes. The default is ``6``.
        max_delta_s : float, optional
            Delta S convergence criteria. The default is ``0.02``.
```

File: pyaedt/modules/SolveSetup.py [hunk 8]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        freq : float, str
+            Frequency at which set the adaptive convergence. It can be float (GHz) or str.
+        max_passes : int
+            Maximum number of adaptive passes.
+        max_delta_s : float
+            Delta S Convergence criteria.
+
+        Returns
+        -------
+        bool
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:557]:
```suggestion
        bool
            ``True`` when successful, ``False`` when failed.
```

File: pyaedt/modules/SolveSetup.py [hunk 9]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        high_frquency : float, str
+            Lower Frequency at which set the adaptive convergence. It can be float (GHz) or str.
+        max_passes : int
+            Maximum number of adaptive passes.
+        max_delta_s : float
+            Delta S Convergence criteria.
+
+        Returns
+        -------
+        bool
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:592]:
```suggestion
        bool
            ``True`` when successful, ``False`` when failed.
```

File: pyaedt/modules/SolveSetup.py [hunk 10]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        """
+        if self.setuptype != 1:
+            self._app.logger.error("Method applies only to Hfss Driven Solutions.")
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:595]:
```suggestion
            self._app.logger.error("Method applies only to HFSS-driven solutions.")
```

File: pyaedt/modules/SolveSetup.py [hunk 11]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        self.props["MultipleAdaptiveFreqsSetup"]["Low"] = low_frequency
+        self.props["MultipleAdaptiveFreqsSetup"]["High"] = high_frquency
+        self.props["MaximumPasses"] = max_passes
+        self.props["MaxDeltaS"] = max_delta_s
+        self.auto_update = True
+        return self.update()
+
+    @pyaedt_function_handler()
+    def enable_adaptive_setup_multifrequency(self, frequencies, max_delta_s=0.02):
+        """Enable Hfss Multifrequency Setup.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:616]:
```suggestion
        """Enable HFSS multi-frequency setup.
```

File: pyaedt/modules/SolveSetup.py [hunk 12]
```diff
@@ -538,6 +538,115 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        Parameters
+        ----------
+        frequencies : list
+            Frequency at which set the adaptive convergence. List entries can be float (GHz) or str.
+        max_delta_s : list or float
+            Delta S Convergence criteria.
+
+        Returns
+        -------
+        bool
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:627]:
```suggestion
        bool
            ``True`` when successful, ``False`` when failed.
```

File: pyaedt/modules/SolveSetup.py [hunk 13]
```diff
@@ -538,6 +538,122 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+
+        Parameters
+        ----------
+        low_frequency : float, str
+            Lower Frequency at which set the adaptive convergence.
+            It can be float (GHz) or str.
+        high_frquency : float, str
+            Lower Frequency at which set the adaptive convergence. It can be float (GHz) or str.
+        max_passes : int
+            Maximum number of adaptive passes.
```
Comment (MaxJPRey) [inline pyaedt/modules/SolveSetup.py:589]:
```suggestion
            Maximum number of adaptive passes. The default is ``6``.
```

File: pyaedt/modules/SolveSetup.py [hunk 14]
```diff
@@ -538,6 +538,122 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        max_delta_s : float
+            Delta S Convergence criteria.
```
Comment (MaxJPRey) [inline pyaedt/modules/SolveSetup.py:591]:
```suggestion
            Delta S Convergence criteria. The default is ``0.02``.
```

File: pyaedt/modules/SolveSetup.py [hunk 15]
```diff
@@ -538,6 +538,122 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        Parameters
+        ----------
+        low_frequency : float, str
+            Lower Frequency at which set the adaptive convergence.
+            It can be float (GHz) or str.
+        high_frquency : float, str
+            Lower Frequency at which set the adaptive convergence. It can be float (GHz) or str.
+        max_passes : int
+            Maximum number of adaptive passes.
+        max_delta_s : float
```
Comment (MaxJPRey) [inline pyaedt/modules/SolveSetup.py:590]:
```suggestion
        max_delta_s : float, optional
```

File: pyaedt/modules/SolveSetup.py [hunk 16]
```diff
@@ -538,6 +538,122 @@ def add_eddy_current_sweep(self, range_type="LinearStep", start=0.1, end=100, co
...
+        """Enable HFSS broadband setup.
+
+        Parameters
+        ----------
+        low_frequency : float, str
+            Lower Frequency at which set the adaptive convergence.
+            It can be float (GHz) or str.
+        high_frquency : float, str
+            Lower Frequency at which set the adaptive convergence. It can be float (GHz) or str.
+        max_passes : int
```
Comment (MaxJPRey) [inline pyaedt/modules/SolveSetup.py:588]:
```suggestion
        max_passes : int, optional
```

File: pyaedt/modules/SolveSetup.py (hunk 1 above)
Comment (soaringkoala) [reply pyaedt/modules/SolveSetup.py:610]: is it still "high_frquency" in version 0.11? when i use "high_frequency", hfss raise error like this. when i switch it to "high_frquency", it worked
image

**Pull request merged**